cli/src/pcluster/models/imagebuilder.py

```diff
@@ -185,24 +207,23 @@ def bucket(self):
             service_name=self.image_name,
             stack_name=self.image_name,
             custom_s3_bucket=custom_bucket_name,
-            artifact_directory=self.s3_artifacts_dir,
+            artifact_directory=self.s3_artifact_dir,
         )
         return self.__bucket
 
     def _get_custom_bucket(self):
         """Try to get custom bucket name from image tag or stack tag."""
         custom_bucket_name = None
         try:
-            image_info = AWSApi.instance().ec2.describe_image_by_name_tag(self.image_name)
-            custom_bucket_name = search_tag(image_info, PCLUSTER_S3_BUCKET_TAG)
-        except AWSClientError as e:
-            if not isinstance(e, ImageNotFoundError):
+            custom_bucket_name = self.image.s3_bucket_name
+        except ImageBuilderActionError as e:
+            if f"Image {self.image_name} does not exist." not in str(e):
```

Comment:
Here you can catch NonExistingImageError

Reply:
done